### ISSUE TITLE
Proposal: Make Hex Editor the default editor for binary files

### ISSUE THREAD

**Issue by SetTrend** [microsoft/vscode-hexeditor#387]
Even after installing the Hex Editor extension, VS Code still keeps nagging me with a warning claiming that the file cannot be opened when I try to inspect a file by double-clicking it in the Explorer:
![Open hex editor by default](https://user-images.githubusercontent.com/9283914/202463020-8a9b0edc-33fd-4605-a7e6-6fc46d28d968.png)

Please, make Hex Editor the default editor for any file, regardless of file extension, that would otherwise cause this warning to appear.

**Comment by lramos15** [comment]
Add this to your settings `"workbench.editor.defaultBinaryEditor": "hexEditor.hexedit",`

**Comment by SetTrend** [comment]
Thank you, @iramos15, for pointing me!

Yet, shouldn't the extension installation script rather change this setting this by default?

**Comment by SetTrend** [comment]
Yes, I see … Though, wouldn't it make sense to change this setting if _undefined_?

If there is no other editor set, then why not set it then? You know, that's why the Hex Editor is installed in the first place.

The extension could peek for an editor defined in workspace settings. If none is set, Hex Editor should set itself. And it should remove itself if set in VS Code's workspace settings when being uninstalled again.

**Comment by lramos15** [comment]
> Yes, I see … Though, wouldn't it make sense to change this setting if _undefined_?
> 
> If there is no other editor set, then why not set it then? You know, that's why the Hex Editor is installed in the first place.
> 
> The extension could peek for an editor defined in workspace settings. If none is set, Hex Editor should set itself. And it should remove itself if set in VS Code's workspace settings when being uninstalled again.

If you link the link in the warning it should prompt you with a set of installed editors to use therefore no default often makes sense if you have a few editors that support binary files

**Comment by SetTrend** [comment]
> no default often makes sense if you have a few editors that support binary files

_"have **a few** editors"_ is key here. If you don't (which is the default for VS Code), a default would still make sense, though. That's exactly what I wrote in my previous comment.

### Pseudo code:
```
if (config["workbench.editor.defaultBinaryEditor"] === null)
  config.write("workbench.editor.defaultBinaryEditor", "hexEditor.hexedit", ConfigScope.Workspace);
```
